### ISSUE TITLE
Heads up! Windows installation even without WSL

### ISSUE THREAD

**Issue by kimsama** [stitionai/devika#155]
Hey, guys
Look this guy, I found this morning. He successfully installed Devika on Windows. 
https://www.youtube.com/watch?v=gcMdzGrDLlw&t=765s

Weird, not much different to mine but tried on `conda` env.  
Please, leave a message if anyone gets success with followinig this.  

Cheers,

**Comment by javacaliente** [comment]
This other guy also got it to work via conda.  https://www.youtube.com/watch?v=eLiMpEIRBzY
Me via WSL and uv, not so much.

**Comment by dongle132** [comment]
Have it running without WSL but only on test VM - little bit try and error (like "pip install vite")  :

![image](https://github.com/stitionai/devika/assets/31986934/774f517c-6cfa-4ccc-adf4-9103438b4d7f)

**Comment by NekoMaster3000** [comment]
I also managed to run it on Windows without any significant changes. (just using the windows prerelease of Bun)

**Comment by PxlNote** [comment]
Are you using Bing or Google. havent seen anyone with bing search have any Problems so far

**Comment by dongle132** [comment]
BING Search API ... just created an account and retrieved the API Key:
![image](https://github.com/stitionai/devika/assets/31986934/0774717b-92f3-4dd8-878e-bd60ff0305c7)

**Comment by 01101010110** [comment]
Hi I finally got it installed in Windows and wrote out a small guide if it helps anyone, along with sources:

DEPENDENCIES:

POWERSHELL:
All code shared below should be run in Powershell as Admin.

PYTHON: 
https://www.python.org/downloads/

OLLAMA:
https://ollama.com/download/windows

GIT:
https://gitforwindows.org/

BUN (You Must Close and Reopen Powershell before running the INSTALL CODE for bun to work):
`powershell -c "irm bun.sh/install.ps1|iex"  `

NODEJS (You Must Close and Reopen Powershell before running the INSTALL CODE for npm to work):
`Set-ExecutionPolicy Bypass -Scope Process -Force; [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1')); choco install nodejs -y`

INSTALL CODE (Paste the whole code at once):
`cd C:\; git clone https://github.com/stitionai/devika.git; cd C:\devika; python -m venv env; env\Scripts\activate; pip install -r requirements.txt; playwright install; python -m playwright install-deps; cd C:\devika\ui; npm install; bun install; $WScriptShell = New-Object -ComObject WScript.Shell; $Shortcut = $WScriptShell.CreateShortcut([System.IO.Path]::Combine([System.Environment]::GetFolderPath([System.Environment+SpecialFolder]::DesktopDirectory), "Backend.lnk")); $Shortcut.TargetPath = "powershell.exe"; $Shortcut.Arguments = "-NoExit -Command `"`& {cd C:\devika; env\Scripts\activate; python devika.py}`""; $Shortcut.Save(); $WScriptShell = New-Object -ComObject WScript.Shell; $Shortcut = $WScriptShell.CreateShortcut([System.IO.Path]::Combine([System.Environment]::GetFolderPath([System.Environment+SpecialFolder]::DesktopDirectory), "Frontend.lnk")); $Shortcut.TargetPath = "powershell.exe"; $Shortcut.Arguments = "-NoExit -Command `"`& {cd C:\devika; env\Scripts\activate; cd ui; bun run dev}`""; $Shortcut.Save()`

ADD KEYS (this will open the config in notepad so you can add your keys and then save):
`Start-Process notepad C:\devika\config.toml -Verb runAs
`
Now run the Backend and Frontend shortcuts on your desktop then open localhost:3000 in your browser.

If the shortcuts dont work you may need to run this command in Powershell as admin then try again:
`Set-ExecutionPolicy RemoteSigned
`
The terminal does not show movement for most users in windows, though Devika is actually working. 
Check C:\devika\projects to find your files and for best results right now, specify a programming language like python 

To run python projects, navigate to C:\devika\projects\yourprojectname:
```
python -m venv env
pip install -r requirements.txt
then run python yourprogramname.py

```

Sources:
Devika Github page
https://github.com/stitionai/devika

OpenAI’s API Key page:
https://platform.openai.com/api-keys

Bing's Web Search API page:
https://www.microsoft.com/en-us/bing/apis/bing-web-search-api

The two videos below combined taught me enough that I could build this guide and make the steps a little easier for Windows users.

Introducing DEVIKA - OpenSource AI Software Engineer | Local Install -- this video shows us the forked version to fix websearch
https://youtu.be/eLiMpEIRBzY?si=FFosaHr1baYQBtmo&t=177

Install Devika Locally on Windows - Agentic AI Software Engineer
https://www.youtube.com/watch?v=f5xYrAgqRAo&t=329s

This one explains the bing web api 

Make Applications using Free AI Software Engineer on Your Computer
https://www.youtube.com/watch?v=BIGs_yGpZNQ&t=155s

DEVIKA | Getting Started [A-Z] Install & Run | Use Cases
https://www.youtube.com/watch?v=gcMdzGrDLlw&t=765s

The one to finally show us -- the python env was needed
Trying DEVIKA (Opensource AI Software Engineer) in Local | Devin alternative
https://www.youtube.com/watch?v=w9QQ39qN0jY

**Comment by 01101010110** [comment]
Hi I finally got it working and wrote out a little guide of the steps that worked for me, maybe it will help someone else?

DEPENDENCIES:

POWERSHELL:
All code shared below should be run in Powershell as Admin.

PYTHON: 
https://www.python.org/downloads/

OLLAMA:
https://ollama.com/download/windows

GIT:
https://gitforwindows.org/

BUN (You Must Close and Reopen Powershell before running the INSTALL CODE for bun to work):
```powershell -c "irm bun.sh/install.ps1|iex"  ```

NODEJS (You Must Close and Reopen Powershell before running the INSTALL CODE for npm to work):
```Set-ExecutionPolicy Bypass -Scope Process -Force; [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1')); choco install nodejs -y```

INSTALL CODE:
```cd C:\; git clone https://github.com/stitionai/devika.git; cd C:\devika; python -m venv env; env\Scripts\activate; pip install -r requirements.txt; playwright install; python -m playwright install-deps; cd C:\devika\ui; npm install; bun install; $WScriptShell = New-Object -ComObject WScript.Shell; $Shortcut = $WScriptShell.CreateShortcut([System.IO.Path]::Combine([System.Environment]::GetFolderPath([System.Environment+SpecialFolder]::DesktopDirectory), "Backend.lnk")); $Shortcut.TargetPath = "powershell.exe"; $Shortcut.Arguments = "-NoExit -Command `"`& {cd C:\devika; env\Scripts\activate; python devika.py}`""; $Shortcut.Save(); $WScriptShell = New-Object -ComObject WScript.Shell; $Shortcut = $WScriptShell.CreateShortcut([System.IO.Path]::Combine([System.Environment]::GetFolderPath([System.Environment+SpecialFolder]::DesktopDirectory), "Frontend.lnk")); $Shortcut.TargetPath = "powershell.exe"; $Shortcut.Arguments = "-NoExit -Command `"`& {cd C:\devika; env\Scripts\activate; cd ui; bun run dev}`""; $Shortcut.Save()```

ADD KEYS (this will open the config in notepad so you can add your keys and then save):
```Start-Process notepad C:\devika\config.toml -Verb runAs```

Now run the Backend and Frontend shortcuts on your desktop then open ```localhost:3000``` in your browser.

If the shortcuts dont work you may need to run this command in Powershell as admin then try again:
```Set-ExecutionPolicy RemoteSigned```

The terminal does not show movement for most users in windows, though Devika is actually working. 
Check C:\devika\projects to find your files and for best results right now, specify a programming language like python 

To run python projects, navigate to C:\devika\projects\yourprojectname:
```python -m venv env
pip install -r requirements.txt
then run python yourprogramname.py```

Video of the steps if needed:
https://youtu.be/bvFBEDbbwFQ

**Comment by Ahmet0691** [comment]
terminal part is not working, what is the reason and what to do?

**Comment by NAME0x0** [comment]
The installation was tough but has worked. I have but just one problem now. Devika is interactive but not functional. I can click on the "New Project" button but when I type in the name of the project and click ok or press enter, nothing happens. I can't choose a model and no model is even being displayed in the dropdown list. Same thing with the Search Engine as well. If I go to the settings area, nothing is visible, exactly the same with history as well. No issue coming up on the backend nor the frontend parts. Tried multiple clean reinstalls of the repo. Tried conda and uv. Conda seems more convenient but I am working with uv right now. 


![Opera Snapshot_2024-04-08_021059_localhost](https://github.com/stitionai/devika/assets/119836361/854e89a0-f98d-459d-8715-b0dd21e57256)


![Opera Snapshot_2024-04-08_021232_localhost](https://github.com/stitionai/devika/assets/119836361/5a558859-db12-4fd3-9a19-eee022854da7)


![Opera Snapshot_2024-04-08_021303_localhost](https://github.com/stitionai/devika/assets/119836361/fd4f2834-9fda-428d-a896-b76817e70a49)


![Screenshot (4)](https://github.com/stitionai/devika/assets/119836361/e2e3efd7-c09d-49f1-b0ac-ca5bb668b0cf)


![Screenshot (5)](https://github.com/stitionai/devika/assets/119836361/de90c8e2-5c7f-4c25-b0f5-1ad332e4e848)

**Comment by javacaliente** [comment]
"The installation was tough but has worked."
Which set of instructions?  There's more than 1.

**Comment by NAME0x0** [comment]
> "The installation was tough but has worked." Which set of instructions? There's more than 1.

I first replicated the steps as shown on this YouTube video but for Windows: [https://youtu.be/kw9nTK42bTw?si=cEXquDJ0ng5S9Ua8](url)
and then I replicated the installation method on this other YouTube video: [https://youtu.be/gcMdzGrDLlw?si=o7-v_sZr7JjTxWUH](url)
and finally I just repeated the method as shown by [01101010110] in this issue. The key point to note is that installation worked in each case but I found [01101010110]'s method to be really convenient and something more comfortable to work with as there were literally only 5 steps. However, in each case the result was similar. The UI comes up and is interactive but not functional.
